src/unzip.ts

```diff
@@ -56,18 +83,21 @@ const extractEntriesFromBuffer = async (data: Buffer, outputDir: string): Promis
 
     logger.debug(`[unzip] Processing entry: ${entry.fileName}`);
     if (entry.fileName.endsWith(path.sep)) {
+      logger.debug(`[unzip] mkdir: ${outputFilePath}`);
       await fs.promises.mkdir(outputFilePath, { recursive: true });
     } else {
       const parentDir = outputFilePath.substring(0, outputFilePath.lastIndexOf(path.sep));
+      logger.debug(`[unzip] else mkdir: ${parentDir}`);
       await fs.promises.mkdir(parentDir, { recursive: true });
 
       const compressionMethod = entryHeader.readUInt16LE(8);
-      if (compressionMethod === 0) {
+      if (entry.compressedSize === 0 || compressionMethod === 0) {
```

Comment:
I don't love that we're using the compressedSize to make decisions - though that might legitimately be part of the header. Do you have a link to the header breakdown (or the bitFlag === 8 details) we could leave as a comment?

Reply:
Good catch - this actually does not need to be here, was leftover from a different approach i tried.